rcl/src/rcl/service.c

```diff
@@ -203,6 +203,25 @@ rcl_send_response(
   return RCL_RET_OK;
 }
 
+bool
+rcl_service_is_valid(const rcl_service_t * service)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(service, false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(service->impl,
+                              "rcl service implementation is invalid",
+                              return false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(service->impl->rmw_handle,
+                              "rmw handle is invalid",
+                              return false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(service->impl->rmw_handle->service_name,
+                              "service name is invalid",
+                              return false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(*service->impl->rmw_handle->service_name,
+                              "service name is invalid",
+                              return false);							  
```

Comment:
Also, why check the service name here, but not check the topic name in publisher and subscription? I'm not sure you need to check it anywhere. I don't see any way that it could end up invalid.